### PR TITLE
New upgrades

### PR DESCRIPTION
**Pull request by mes258**

### CODE REVIEW

File: tower.js [hunk 1]
```diff
@@ -497,4 +490,31 @@ CollectorTower.prototype.extractResource = function(resource, state){
     state.uranium += this.extractRate;
   }
   resource.extract(this.extractRate);
+}
+
+
+function getNewTower(type, pos, state){
+  switch(type){
```
Comment (vinport16) [inline tower.js:497]: hm should have added me as a reviewer lol
this should be a map object and not a big case/switch it is making me uncomfortable :(

Comment (vinport16) [reply tower.js:497]: storing the name of the object as a string and then using that string to find the object later ???? what is this

Comment (vinport16) [reply tower.js:497]: i'm going to report you to liberatore and he will retroactively fail you for this,

Comment (mes258) [reply tower.js:497]: Most of the changes in this PR were from 2020. A lot of refactoring is needed at some point for this project.

**Pull request merged**